### DISCUSSION TITLE
Photo's not being imported

### DISCUSSION THREAD

**Discussion by eljejer** [photoprism/photoprism#3279] [Q&A] [answered]
I cannot find out why the items in the import directory are not shown in originals NOR are they removed from import after the import.
The import volume is declared in yml: "~/PhotoPrism/Import:/photoprism/import"
At the moment there are a couple of jpg, jpeg and png files in the import folder
When I do "All files from import folder"t and the "Move Files" is checked, I see this in the log:

2023-03-13 09:53:51 INFO import completed in 1 s

2023-03-13 09:53:50 INFO import: imported 1 file

2023-03-13 09:53:50 INFO import: found no .ppignore file

2023-03-13 09:53:50 INFO moving files from import

But then, all the files remain in the import folder and I cannot find any of them in 'originals'
I wanted to get a feeling on the workflow of 'import', but like this it's not getting me there -:) 
Anyone???
PhotoPrism+ build is Build 230311-3465d0e34
OS is Linux

**Comment by graciousgrey** [comment]
Might be permission issues or some of the other reasons listed here:

https://docs.photoprism.app/getting-started/troubleshooting/#missing-pictures

Turning on debug logs might give additional hints.

**Reply by eljejer** [reply]
Initially I had been using photoprism as root. Now I have changed the configuration to start photoprism as non-root user by changing PHOTOPRISM_UID in docker-compose.yml: 

`PHOTOPRISM_UID: 33`

(user 33 is www-data).
And I have added this user to group 'users':

```
$ id www-data
uid=33(www-data) gid=33(www-data) groups=33(www-data),100(users)
```

The Permissions on the import directory are:

```
$ ls -la
total 44
drwxrwxr-x  5 eljejer users    4096 mrt 15 14:23 .
drwxr-x--- 40 eljejer eljejer  4096 mrt 15 14:22 ..
drwxr-xr-x  6     999 docker   4096 mrt 15 14:20 database
-rw-rw-r--  1 eljejer eljejer 12051 mrt 15 14:23 debug.log
-rwxrwxrw-  1 eljejer users    9056 mrt 14 23:10 docker-compose.yml
drwxrwxr-x  2 eljejer users    4096 mrt 13 20:42 Import
drwxrwxrw-  8 eljejer users    4096 mrt 12 07:56 storage
drwxrwxr-x  2 eljejer users   4096 mrt 13 20:42 Import.

```

This way I would expect photoprism to be able to move files from the import directory to 'originals'.
[debug.log](https://github.com/photoprism/photoprism/files/10980443/debug.log)

I attach debug log from the moment from start - import - end

**Comment by eljejer** [comment]
Hello again

So I'm still struggling to be able to use the import functionality of photoprism.
I have repeated the steps with  PHOTOPRISM_LOG_LEVEL: "debug" in the environment of photoprism service.
See debug.log in previous comment

B.t.w., when I do an archive of photo's and thereafter a delete of these, then the files in question ARE deleted from the filesystem also. So, no permission issues here.

**Comment by eljejer** [comment, marked as answer]
Hmm, I don't know what changed in the setup, but right now, I have no problems with the import functionality

**Comment by williamtrelawny** [comment]
I have this same issue. Turned on trace level logs, nothing indicating any permissions or any other issue at all. 

Is there a definitive answer to what exactly the permissions of all folders should be for PhotoPrism to work? I'm running in root mode and with the Docker Compose installation method.

OS: Debian 12.4
PhotoPrism version: Build 231128-f48ff16ef
MariaDB version: 11.2.2-MariaDB-1:11.2.2+maria~ubu2204
Docker version info:
```
Client: Docker Engine - Community
 Version:           24.0.7
 API version:       1.43
 Go version:        go1.20.10
 Git commit:        afdd53b
 Built:             Thu Oct 26 09:08:02 2023
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          24.0.7
  API version:      1.43 (minimum version 1.12)
  Go version:       go1.20.10
  Git commit:       311b9ff
  Built:            Thu Oct 26 09:08:02 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.26
  GitCommit:        3dd1e886e55dd695541fdcd67420c2888645a495
 runc:
  Version:          1.1.10
  GitCommit:        v1.1.10-0-g18a0cb0
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Reply by williamtrelawny** [reply]
Nevermind. I just discovered that PhotoPrism is apparently hard-coded to use a folder named `import` for imports, despite what volume path you specify in docker-compose.yml. I had:
```
 - "~/imports:/photoprism/imports"
 ```
but when I ran `ls -l` within the container I saw there was both an `import` and the `imports` folder I mapped in docker-compose.yml. So when I was attempting to run the Import function it didn't find anything of course.

I guess it's not a huge deal, but it was unexpected at least. Guess I wouldn't call it a bug, but figured I'll let this post serve as a note for anyone else who may want to use a different imports folder name (pluralizing it just made more semantical sense to me, plus it matched the pluralization of `originals` as well).

**Reply by lastzero** [reply]
@williamtrelawny Be aware that running the containers as `root` will point the home directory shortcut `~` to `/root` in the `volumes:` section of your `docker-compose.yml`, see the notes in our docs and the config examples:

- https://docs.photoprism.app/getting-started/docker-compose/#step-1-configure
- https://dl.photoprism.app/docker/docker-compose.yml

PhotoPrism also allows you to configure the paths it uses within the Docker container so that they are not really static and could be changed:

- https://docs.photoprism.app/getting-started/config-options/#storage

When using Docker, however, this makes little sense, as the files/folders visible to PhotoPrism depend only on the path you mount from your host, so it makes no difference which path you use within the container :)

**Reply by williamtrelawny** [reply]
So I'm running Docker and the containers as non-root user and I can confirm `~` is pointing to my user's `$HOME` not `/root`. Sorry if I was unclear. 

And yeah, makes little sense. Just one of those defaults I didn't know about. Your all's docs really are some of the best I've seen though so I'm not complaining!
